src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -78,6 +79,10 @@ def set_opts(self, curl, url, params, headers,
         curl.setopt(pycurl.FOLLOWLOCATION, self.followlocation)
         curl.setopt(pycurl.MAXREDIRS, self.maxredirs)
 
+        if params:
+            if  isinstance(params, dict):
+                params = json.dumps(params)
+
         if  verb == 'GET':
             encoded_data = urllib.urlencode(params, doseq=doseq)
```

Comment:
Sorry to hear that. Hope everything is fine with you. Thanks for updating the code.